File: src/main/java/com/example/domain/admin/dto/MemberDetailResponse.java
package com.example.domain.admin.dto;

import com.example.domain.member.entity.Member;

import java.time.LocalDateTime;

public record MemberDetailResponse(
        Long memberId,
        String loginId,
        String email,
        String nickname,
        int reportCount,
        String status,
        String role,
        LocalDateTime createdAt
) {
    public static MemberDetailResponse from(Member member) {
        return new MemberDetailResponse(
                member.getId(),
                member.getLoginId(),
                member.getEmail(),
                member.getNickname(),
                member.getReportCount(),
                member.getStatus().name(),
                member.getRole().name(),
                member.getCreatedAt()
        );
    }
}

File: src/main/java/com/example/domain/admin/dto/response/MemberDetailResponse.kt
package com.example.domain.admin.dto.response

import com.example.domain.member.entity.Member
import java.time.LocalDateTime

data class MemberDetailResponse(
    val memberId: Long,
    val loginId: String,
    val email: String,
    val nickname: String,
    val reportCount: Int,
    val status: String,
    val role: String,
    val createdAt: LocalDateTime?
) {
    companion object {
        @JvmStatic
        fun from(member: Member): MemberDetailResponse {
            return MemberDetailResponse(
                member.id,
                member.loginId,
                member.email,
                member.nickname,
                member.reportCount,
                member.status.name,
                member.role.name,
                member.createdAt
            )
        }
    }
}
